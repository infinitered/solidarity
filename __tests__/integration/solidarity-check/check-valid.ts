import execa from 'execa'
import tempy from 'tempy'

const path = require('path')
const filesystem = require('fs-jetpack')
const SOLIDARITY = `node ${process.cwd()}${path.sep}bin${path.sep}solidarity`
const origCwd = process.cwd()
let originalTimeout

beforeAll(() => {
  // These can be slow on CI
  originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000
})

afterAll(function() {
  // Fix timeout change
  jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout
})

test('default looks for .solidarity file', async done => {
  const tempDir = tempy.directory()
  filesystem.copy(
    `__tests__${path.sep}sandbox${path.sep}solidarity_json${path.sep}.solidarity.json`,
    `${tempDir}${path.sep}.solidarity`
  )
  process.chdir(tempDir)
  try {
    await execa.shell(SOLIDARITY).then(result => {
      expect(result.stdout.trim()).toMatchSnapshot()
      done()
    })
  } catch (err) {
    done.fail()
  }
})

test('also looks for .solidarity.json file', async done => {
  const tempDir = tempy.directory()
  filesystem.copy(
    `__tests__${path.sep}sandbox${path.sep}solidarity_json${path.sep}.solidarity.json`,
    `${tempDir}${path.sep}.solidarity.json`
  )
  process.chdir(tempDir)
  try {
    await execa.shell(SOLIDARITY).then(result => {
      expect(result.stdout.trim()).toMatchSnapshot()
      done()
    })
  } catch (err) {
    done.fail()
  }
})

test('verbose flag works', async done => {
  const tempDir = tempy.directory()
  filesystem.copy(
    `__tests__${path.sep}sandbox${path.sep}solidarity_json${path.sep}.solidarity.json`,
    `${tempDir}${path.sep}.solidarity`
  )
  process.chdir(tempDir)
  try {
    await execa.shell(`${SOLIDARITY} --verbose`).then(result => {
      expect(result.stdout.trim()).toMatchSnapshot()
      done()
    })
  } catch (err) {
    done.fail()
  }
})

test('silent flag works', async done => {
  const tempDir = tempy.directory()
  filesystem.copy(
    `__tests__${path.sep}sandbox${path.sep}solidarity_json${path.sep}.solidarity.json`,
    `${tempDir}${path.sep}.solidarity`
  )
  process.chdir(tempDir)
  try {
    await execa.shell(`${SOLIDARITY} --silent`).then(result => {
      expect(result.stdout.trim()).toMatchSnapshot()
      done()
    })
  } catch (err) {
    done.fail()
  }
})

afterEach(() => {
  process.chdir(origCwd)
})
