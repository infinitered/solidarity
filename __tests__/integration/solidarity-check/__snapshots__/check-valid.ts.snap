// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`also looks for .solidarity.json file 1`] = `"✔︎ Solidarity checks valid"`;

exports[`default looks for .solidarity file 1`] = `"✔︎ Solidarity checks valid"`;

exports[`silent flag works 1`] = `""`;

exports[`verbose flag works 1`] = `"✔︎ Solidarity checks valid"`;
